flood_adapt/object_model/hazard/forcing/rainfall.py

```diff
@@ -111,3 +113,29 @@ def save_additional(self, output_dir: Path | str | os.PathLike) -> None:
     @classmethod
     def default(cls) -> "RainfallCSV":
         return RainfallCSV(path="path/to/rainfall.csv")
+
+
+class RainfallNetCDF(IRainfall):
+    source: ForcingSource = ForcingSource.NETCDF
+    unit: us.UnitTypesIntensity = us.UnitTypesIntensity.mm_hr
+
+    path: Path
+
+    def read(self) -> xr.Dataset:
+        ds = xr.open_dataset(self.path)
+        required_vars = {"precip"}
+        required_coords = {"time", "lat", "lon"}
+        return validate_netcdf_forcing(ds, required_vars, required_coords)
+
+    def save_additional(self, output_dir: Path | str | os.PathLike) -> None:
+        if self.path:
+            output_dir = Path(output_dir)
+            if self.path == output_dir / self.path.name:
```

Comment:
you might want to normalize the path here so that `/foo/bar/` and `/foo/bar/../bar` will end up as the same